File: 1399._Count_Largest_Group_E.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        max_cnt = float("-inf")
        cnt = collections.defaultdict(int)

        for i in range(1, n + 1):
            tmp = 0
            while i:
                tmp += i % 10
                i //= 10
            
            cnt[tmp] += 1
            max_cnt = max(max_cnt, cnt[tmp])

        result = 0
        for val in cnt.values():
            if val == max_cnt:
                result +=1

        return result

File: 1399._Count_Largest_Group_E.go
func countLargestGroup(n int) int {
	max_cnt := 0
	cnt := make(map[int]int)
	var tmp_sum int

	for i := 1; i <= n; i++ {
		tmp_sum = 0
		for ii := i; ii > 0; ii /= 10 {
			tmp_sum += ii % 10
		}
		cnt[tmp_sum]++
		max_cnt = max(max_cnt, cnt[tmp_sum])
	}

	var result int
	for _, value := range cnt {
		if max_cnt == value {
			result++
		}

	}

	return result    
}
